atom-jobs/atom-ut-codecov.groovy

```diff
@@ -0,0 +1,213 @@
+                            command: '/bin/sh -c', args: 'cat',
+                            envVars:[containerEnvVar(key: 'GOPATH', value: '/go')],
+                    ),
+                    containerTemplate(
+                            name: 'jnlp', image: "${jnlp_docker_image}", alwaysPullImage: false,
+                            resourceRequestCpu: '100m', resourceRequestMemory: '256Mi',
+                    ),
+            ],
+            volumes: [
+                    nfsVolume(mountPath: '/home/jenkins/agent/ci-cached-code-daily', serverAddress: '172.16.5.22',
+                            serverPath: '/mnt/ci.pingcap.net-nfs/git', readOnly: false),
+            ],
+    ) {
+        node(label) {
+            println "debug command:\nkubectl -n ${namespace} exec -ti ${NODE_NAME} bash"
+            body()
+        }
+    }
+}
+
+
+lines_coverage_rate = ""
+
+run_with_pod {
+    container("golang") {
+        try {
+            def ws = pwd()
+
+            stage("${TASK_NAME}") {
+                println "${TASK_NAME}"
+            }
+
+            stage("Download binary from fileserver") {
+                sh '''
+                    wget -q -c ${CACHE_CODE_FILESERVER_URL} -O - | tar -xz
+                '''
+            }
+
+            credentialList =[]
+            varStrings = SECRET_VARS.split(",")
+            for (varString in varStrings) {
+                keyAndValue = varString.split(":")
+                if (keyAndValue.length == 2) {
+                    credentialList.push(string(credentialsId: keyAndValue[0], variable: keyAndValue[1]))
+                }
+            }
+
+            stage("Test") {
+                dir("${ws}/${REPO}") {
+                    sh TEST_CMD
+                }
+            }
+
+            stage("Codecov") {
+                // upload coverage report to codecov
+                // read coverage rate from codecov api
+                dir("${ws}/${REPO}") {
+                    // coverage file: test_report/*coverage.out
+                    withCredentials(credentialList) {
+                        timeout(5) {
+                            if (PULL_REQUEST_ID != null && PULL_REQUEST_ID != "") {
+                                sh """
+                                curl -LO ${FILE_SERVER_URL}/download/cicd/ci-tools/codecov
+                                chmod +x codecov
+                                ./codecov -f 'coverage_report/*.coverage'  -t ${CODECOV_TOKEN} -C ${COMMIT_ID} -P ${PULL_REQUEST_ID} -b ${BUILD_NUMBER}
+                                """
+                            } else {
+                                sh """
+                                curl -LO ${FILE_SERVER_URL}/download/cicd/ci-tools/codecov
+                                chmod +x codecov
+                                ./codecov -f 'coverage_report/*.coverage' -t ${CODECOV_TOKEN} -C ${COMMIT_ID} -b ${BUILD_NUMBER} -B ${BRANCH}
```

Comment:
if so, every task should generate cov file like this. We can use params passed in.

Reply:
emm. some debug code here. i will modify here